sentry-android-core/src/main/java/io/sentry/android/core/AndroidOptionsInitializer.java

```diff
@@ -82,11 +101,14 @@ static void init(
   private static void installDefaultIntegrations(
       final @NotNull Context context,
       final @NotNull SentryOptions options,
-      final @NotNull IEnvelopeReader envelopeReader) {
+      final @NotNull IEnvelopeReader envelopeReader,
+      final @NotNull IBuildInfoProvider buildInfoProvider) {
 
     // Integrations are registered in the same order. NDK before adding Watch outbox,
     // because sentry-native move files around and we don't want to watch that.
-    options.addIntegration(new NdkIntegration());
+    if (loadNdkIfAvailable(options, buildInfoProvider)) {
```

Comment:
@bruno-garcia maybe I'd be good to get the `Class<?>` as a return type and pass to `NdkIntegration` as a ctor param, so we do the reflection only once on the class level, wdyt?

Reply:
That makes sense. So the return type would be `@Nullable Class<?>`.
It would also improve things imho if we didn't fiddle around with the options `setEnableNdk` value. Just leave it as-is.
When we check if it's true, if we did succeed in getting the class name through this method, we enable it. Otherwise we don't. It would just be nice of this method didn't have side effects. Simply get the class name if it's possible.

Reply:
yes, but on the NdkIntegration we do need to set setEnableNdk=false because we rely on that to add or not the sentry-android-ndk as a package on the sdk info